esti/wrapped_reader.go

```diff
@@ -0,0 +1,21 @@
+package esti
+
+import "io"
+
+type WrappedReader struct {
+	reader       io.Reader
+	numBytesRead int
+}
+
+func NewWrappedReader(reader io.Reader) *WrappedReader {
+	return &WrappedReader{
+		reader:       reader,
+		numBytesRead: 0,
+	}
+}
+
+func (w *WrappedReader) Read(p []byte) (n int, err error) {
+	n, err = w.reader.Read(p)
+	w.numBytesRead += n
+	return n, err
+}
```

Comment:
Not sure why we need to wrap a reader here.  Couldn't I just implement something like

```go
type CountingReader struct {
	Count int
}

func (r *CountingReader) Read(p []byte) (n int, err error) {
	p[0] = 0
	r.Count += 1
	return 1, nil
}
```
?

Or if I were afraid of buffering working against me, I could replace with
```go
type CountingReader struct {
	Count int
}

func (r *CountingReader) Read(p []byte) (n int, err error) {
	for i := range p {
		p[i] = 0
	}
	r.Count += len(p)
	return len(p), nil
}
```